src/Roslyn.Suppressions/Roslyn.Suppressions/SuppressionChecker.cs

```diff
@@ -117,23 +116,22 @@ public bool IsSuppressed(Diagnostic reportedDiagnostic, string settingsKey)
             return false;
         }
 
-        private static bool IsSameFile(Diagnostic diagnostic, SonarQubeIssue suppressedIssue) =>
+        private static bool IsSameFile(Diagnostic diagnostic, SuppressedIssue suppressedIssue) =>
             diagnostic.Location.SourceTree?.FilePath.EndsWith(suppressedIssue.FilePath, StringComparison.OrdinalIgnoreCase)
             ?? false;
 
-        private static bool IsSameLine(FileLinePositionSpan roslynLineSpan, IssueTextRange sonarTextRange)
+        private static bool IsSameLine(FileLinePositionSpan roslynLineSpan, SuppressedIssue suppressedIssue)
         {
             // Special case: file-level issues
             var roslynFileLevelIssue = IsRoslynFileLevelIssue(roslynLineSpan);
-            if (sonarTextRange == null || roslynFileLevelIssue)
+            var sonarFileLevelIssue = !suppressedIssue.RoslynIssueLine.HasValue;
+            if (sonarFileLevelIssue || roslynFileLevelIssue)
```

Comment:
Minor tweak to the code since we are now checking an `int?` rather than a nullable `IssueTextRange` reference type. The logic is the same, though.